models/user.js

```diff
@@ -0,0 +1,46 @@
+const mongoose = require('mongoose');
+const emailValidator = require('email-validator');
+const bcrypt = require('bcrypt');
+const SALT_ROUNDS = 12;
+
+const UserSchema = new mongoose.Schema(
+  {
+    username: {
+      type: String,
+      required: [true, 'The username is required !'],
+      trim: true,
+      index: { unique: true },
+      minlength: [3, 'The username must contain at least 3 characters !']
+    },
+    email: {
+      type: String,
+      required: [true, 'The email is required'],
+      trim: true,
+      index: { unique: true },
+      lowercase: true,
+      validate: {
+        validator: emailValidator.validate,
+        message: props => `${props.value} is not a valid email address`
+      }
+    },
+    password: {
+      type: String,
+      required: [true, 'The password is required !'],
+      minlength: [8, 'The password must contain at least 8 characters !'],
+      select: false
+    }
+  },
+  { timestamps: true }
+);
+
+UserSchema.pre('save', async function() {
+  const user = this;
```

Comment:
Is there a reason for assigning `this` to a variable, or is it just in pursuit of more readability ?

Reply:
Just for more readability